indexing-service/src/main/java/io/druid/indexing/common/task/AbstractTask.java

```diff
@@ -49,19 +51,23 @@
   @JsonIgnore
   private final TaskResource taskResource;
 
+  //This attribute should be removed eventually as dataSources should be enough
   @JsonIgnore
   private final String dataSource;
```

Comment:
Why not remove it now? We can keep serde support for backwards compatibility but I don't see a reason for the field to be there.